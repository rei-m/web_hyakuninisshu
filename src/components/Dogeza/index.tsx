import * as React from 'react';
import styled from 'styled-components';
import * as dogeza from './dogeza_businessman.png';

const DogezaImg = styled.img`
  width: 200px;
`;

const Dogeza: React.SFC<React.ImgHTMLAttributes<{}>> = ({ alt, style }) => (
  <DogezaImg src={dogeza} alt={alt} style={style} />
);

export default Dogeza;
