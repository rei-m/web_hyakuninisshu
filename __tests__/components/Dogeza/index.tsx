import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { setUpQueryOnce } from '@test/helpers';
import Dogeza, { QueryData } from '@src/components/atoms/Dogeza';

describe('<Dogeza />', () => {
  beforeEach(() => {
    setUpQueryOnce<QueryData>({
      dogezaImage: {
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: `100 200 300`,
            src: `dogeza-base64-encoded-image`,
            srcSet: `asdfasdf`,
          },
        },
      },
    });
  });

  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<Dogeza alt="test" />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
