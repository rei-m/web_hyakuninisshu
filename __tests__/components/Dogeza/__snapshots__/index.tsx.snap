// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dogeza /> should render component 1`] = `
<img
  alt="test"
  className="sc-bdVaJa dxKQNh"
  src="dogeza_businessman.png"
  style={undefined}
/>
`;
